File: ex1.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
    for i := 0; i < len(nums); i++ {
        for k := i+1; k < len(nums); k++ {
            if nums[k] == target - nums[i] {
                return []int{i, k}
            }
        }
    }
    return []int{-1, -1}
}

File: leetcode/ex1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> pair;
        unordered_map<int, int> table;
        
        for (int i = 0; i < nums.size(); i++) {
            if (table.find(target - nums[i]) != table.end()) {
                pair.push_back(i);
                pair.push_back(table[target - nums[i]]);
                break;
            }
            table[nums[i]] = i;
        }
        
        return pair;
    }
};
